R/rank_regression_helpers.R

```diff
@@ -0,0 +1,173 @@
+# Helper function that performs lm():
+lm_ <- function(x,
+                y,
+                distribution,
+                direction = c("x_on_y", "y_on_x")
+) {
+
+  distribution <- two_parametric(distribution)
+  direction <- match.arg(direction)
+
+  # Switch between distributions:
+  switch(distribution,
+         "weibull" = {
+           x <- log(x)
+           y <- qsev(y)
+         },
+         "sev" = {
+           y <- qsev(y)
+         },
+         "lognormal" = {
+           x <- log(x)
+           y <- stats::qnorm(y)
+         },
+         "normal" = {
+           y <- stats::qnorm(y)
+         },
+         "loglogistic" = {
+           x <- log(x)
+           y <- stats::qlogis(y)
+         },
+         "logistic" = {
+           y <- stats::qlogis(y)
+         }
+  )
+
+  if (direction == "x_on_y") {
+    stats::lm(x ~ y)
+  } else {
+    stats::lm(y ~ x)
+  }
+}
+
+
+
+# Approximated confidence intervals for parameters according to Mock:
+conf_mock <- function(dist_params, # loc-scale-parameters
+                      conf_level,
+                      n, # sample size
+                      direction
+) {
+
+  # Check for conf_level:
+  if (!(conf_level %in% c(0.9, 0.95, 0.99))) {
+    stop(
+      "'conf_level' must be 0.90, 0.95 or 0.99",
+      call. = FALSE
+    )
+  }
+
+  # Lookup:
+  conf <- as.character(conf_level)
+  mock_val <- c("0.9" = 1.4, "0.95" = 2, "0.99" = 3.4)[[conf]]
```

Comment:
Would it be possible to interpolate in some way for values between 0.9 and 0.99?

Reply:
Unfortunately not.